source_handlers.go

```diff
@@ -18,12 +18,29 @@ var getSourceDao func(c echo.Context) (dao.SourceDao, error)
 
 func getSourceDaoWithTenant(c echo.Context) (dao.SourceDao, error) {
 	tenantId, err := getTenantFromEchoContext(c)
+	if err != nil {
+		return nil, err
+	}
 
+	userID, err := getUserFromEchoContext(c)
 	if err != nil {
 		return nil, err
 	}
 
-	return dao.GetSourceDao(&dao.SourceDaoParams{TenantID: &tenantId}), nil
+	var user m.User
+	result := dao.DB.Model(&m.User{}).Where("tenant_id = ? and user_id = ?", tenantId, userID).Find(&user)
```

Comment:
This is going to get moved to a `UserDAO` eventually I assume? I also would say this check should happen in a middleware much like `middleware.Tenancy`.

Reply:
I moved it to UserDAO.

yes but maybe we should move to DAO layer later.

To be able pass always any value of UserID to `dao.GetSourceDao` through `SourceDaoParams` I need to change `SourceDaoParams#UserID` to string and populate it from header.

Thanks to that there will be always `SourceDaoParams#UserID` from header.

With this I can fetch "our" user ID (`users#id`) from users table (if there will be present) and then we do filtering logic - it will be somewhere [here](https://github.com/RedHatInsights/sources-api-go/pull/360/commits/62d126133eb3f9bf4cfd00d720f63e79bc5588ca) which is based on your [work](https://github.com/RedHatInsights/sources-api-go/pull/343).


Does this sound OK to you ?